### PR TITLE
Medical Adjustments and fixes

### PR DESCRIPTION
**Pull request by Glowbal**
- Fixes for wierdness in damage handling for basic medical.
- Fixes for updating base classes since the A3 1.50 update.
- Fixes for unconscious waking up in basic medical.
- Removed duplicate code in the damage handling framework
- Clean up of the medical module by @jokoho48 
- Minor bug fixes to various parts of the module.

### CODE REVIEW

File: addons/medical/CfgVehicles.hpp [hunk 1]
```diff
@@ -529,10 +658,55 @@ class CfgVehicles {
 
     class B_Soldier_04_f: B_Soldier_base_F {
         class HitPoints: HitPoints {
-            class HitHead: HitHead {};
-            class HitBody: HitBody {};
-            class HitHands: HitHands {};
-            class HitLegs: HitLegs {};
+            class HitFace: HitHead {
```
Comment (jokoho48) [inline addons/medical/CfgVehicles.hpp]: could it be better for making a this into a macro?

File: addons/medical/functions/fnc_actionLoadUnit.sqf [hunk 2]
```diff
@@ -7,20 +7,18 @@
...
-private ["_caller", "_target","_vehicle"];
-_caller = _this select 0;
-_target = _this select 1;
+private "_vehicle";
+params ["_caller", "_target"];
 
 if ([_target] call EFUNC(common,isAwake)) exitwith {
-    // TODO localization
-    ["displayTextStructured", [_caller], [["This person (%1) is awake and cannot be loaded", [_target] call EFUNC(common,getName)], 1.5, _caller]] call EFUNC(common,targetEvent);
+    ["displayTextStructured", [_caller], [[localize LSTRING(CanNotLoaded), [_target] call EFUNC(common,getName)], 1.5, _caller]] call EFUNC(common,targetEvent);
```
Comment (jokoho48) [inline addons/medical/functions/fnc_actionLoadUnit.sqf]: if i see this right you can use only LSTRING() without localize

File: addons/medical/functions/fnc_actionRemoveTourniquet.sqf [hunk 3]
```diff
@@ -26,7 +24,7 @@ _tourniquets = _target getvariable [QGVAR(tourniquets), [0,0,0,0,0,0]];
 
 // Check if there is a tourniquet on this bodypart
 if ((_tourniquets select _part) == 0) exitwith {
-    _output = "There is no tourniquet on this body part!";
+    _output = localize LSTRING(noTourniquetOnBodyPart);
```
Comment (jokoho48) [inline addons/medical/functions/fnc_actionRemoveTourniquet.sqf]: same as above

File: addons/medical/functions/fnc_addToLog.sqf [hunk 4]
```diff
@@ -9,27 +9,26 @@
...
+    [_this, QFUNC(addToLog), _unit] call EFUNC(common,execRemoteFnc); /* TODO Replace by event system */
 };
 
-_lastNumber = date select 4;
-_moment = format["%1:%2",date select 3, _lastNumber];
-if (_lastNumber < 10) then {
-    _moment = format["%1:0%2",date select 3, _lastNumber];
+date params ["", "", "", "_minute", "_hour"];
+
+_moment = if (_minute < 10) then {
```
Comment (jokoho48) [inline addons/medical/functions/fnc_addToLog.sqf]: posible to use this?
`_moment = format [ (["%1:%2", "%1:0%2"] select (_minute < 10)), _hour, _minute]`

File: addons/medical/functions/fnc_canTreat.sqf [hunk 5]
```diff
@@ -16,17 +16,16 @@
...
+if !(_target isKindOf "CAManBase") exitWith { false };
 
-_config = (ConfigFile >> "ACE_Medical_Actions" >> "Basic" >> _className);
-if (GVAR(level)>=2) then {
-    _config = (ConfigFile >> "ACE_Medical_Actions" >> "Advanced" >> _className);
+
+_config = if (GVAR(level)>=2) then {
+    (ConfigFile >> "ACE_Medical_Actions" >> "Advanced" >> _className)
+} else {
+    (ConfigFile >> "ACE_Medical_Actions" >> "Basic" >> _className)
```
Comment (jokoho48) [inline addons/medical/functions/fnc_canTreat.sqf]: posible use of this _config = (ConfigFile >> "ACE_Medical_Actions" >> (["Basic", "Advanced"] select (GVAR(level)>=2)) >> _className)

File: addons/medical/functions/fnc_displayPatientInformation.sqf [hunk 6]
```diff
@@ -17,10 +18,8 @@
...
 // Exit for basic medical
 if (GVAR(level) < 2) exitWith {};
-private ["_target", "_show", "_selectionN", "_amountOfGeneric", "_bandagedwounds", "_logCtrl", "_part", "_partText", "_pointDamage", "_severity", "_total", "_totalIvVolume", "_triageStatus", "_type"];
-_target = _this select 0;
-_show = if (count _this > 1) then {_this select 1} else {true};
-_selectionN = if (count _this > 2) then {_this select 2} else {0};
+private ["_amountOfGeneric", "_bandagedwounds", "_logCtrl", "_part", "_partText", "_pointDamage", "_severity", "_total", "_totalIvVolume", "_triageStatus", "_type"];
+params ["_target", ["_show", true], ["_selectionN", 0]];
 
 GVAR(currentSelectedSelectionN) = if (typeName _selectionN == "SCALAR") then {_selectionN} else {0};
```
Comment (jokoho48) [inline addons/medical/functions/fnc_displayPatientInformation.sqf]: Posible use this
```GVAR(currentSelectedSelectionN) = [0, _selectionN] select (IS_SCALAR(_selectionN));
GVAR(displayPatientInformationTarget) = [ObjNull, _target] select _show;```

File: addons/medical/functions/fnc_displayPatientInformation.sqf [hunk 7]
```diff
@@ -17,10 +18,8 @@
...
 GVAR(displayPatientInformationTarget) = if (_show) then {_target} else {ObjNull};
```
Comment (jokoho48) [inline addons/medical/functions/fnc_displayPatientInformation.sqf]: Posible use this
```GVAR(currentSelectedSelectionN) = [0, _selectionN] select (IS_SCALAR(_selectionN));
GVAR(displayPatientInformationTarget) = [ObjNull, _target] select _show;```

File: addons/medical/functions/fnc_getTypeOfDamage.sqf [hunk 8]
```diff
@@ -13,23 +13,19 @@
...
-    case (_typeOfProjectile isKindOf "RocketBase"): {"explosive"};
-    case (_typeOfProjectile isKindOf "MissileBase"): {"explosive"};
-    case (_typeOfProjectile isKindOf "LaserBombCore"): {"explosive"};
-    case (_typeOfProjectile isKindOf "BombCore"): {"explosive"};
-    case (_typeOfProjectile isKindOf "Grenade"): {"grenade"};
-    default {toLower _typeOfProjectile};
-};
-
-_typeOfDamage
+call {
```
Comment (jokoho48) [inline addons/medical/functions/fnc_getTypeOfDamage.sqf]: call can removed here

File: addons/medical/functions/fnc_handleDamage.sqf [hunk 9]
```diff
@@ -35,53 +31,46 @@ if (typeName _projectile == "OBJECT") then {
...
 // Exit if we disable damage temporarily
-_damageOld = damage _unit;
-if (_selection in GVAR(SELECTIONS)) then {
-    _damageOld = _unit getHit _selection;
+if !(_unit getVariable [QGVAR(allowDamage), true]) exitWith {
+    _damageOld = damage _unit;
+    if (_selection in GVAR(SELECTIONS)) then {
+        _damageOld = _unit getHit _selection;
+    };
+    _damageOld;
```
Comment (jokoho48) [inline addons/medical/functions/fnc_handleDamage.sqf]: Posible Use of this
```
if !(_unit getVariable [QGVAR(allowDamage), true]) exitWith {
    if (_selection in GVAR(SELECTIONS)) then {
        _unit getHit _selection
    } else {
        damage _unit
    };
};
```

File: addons/medical/functions/fnc_handleDamage.sqf [hunk 10]
```diff
@@ -35,53 +31,46 @@ if (typeName _projectile == "OBJECT") then {
...
-    if (vehicle _unit != _unit && {!(vehicle _unit isKindOf "StaticWeapon")} && {isNull _shooter} && {_projectile == ""} && {_selection == ""}) then {
-        if (GVAR(enableVehicleCrashes)) then {
-            _selection = GVAR(SELECTIONS) select (floor(random(count GVAR(SELECTIONS))));
-        };
+_newDamage = _this call FUNC(handleDamage_caching);
+// handleDamage_caching may have modified the projectile string
+_typeOfDamage = [_projectile] call FUNC(getTypeOfDamage);
+
+_typeIndex = (GVAR(allAvailableDamageTypes) find _typeOfDamage);
+_minLethalDamage = 0.01;
```
Comment (jokoho48) [inline addons/medical/functions/fnc_handleDamage.sqf]: posible use of this 
```
_minLethalDamage = if (_typeIndex >= 0) then {
    GVAR(minLethalDamages) select _typeIndex
} else {
    0.01
};
```

**Comment by Glowbal** [comment]
Closing in favour of #2361

**Pull request closed without merging**